src/diagram/components/ui/variable-chip-list-container.tsx

```diff
@@ -1,28 +1,19 @@
-import React, { useState } from "react";
+import React from "react";
 
 import { VariableChipList } from "./variable-chip-list";
 import { VariableType } from "../../models/variable";
+import { useSelectMultipleVariables } from "../../../hooks/use-select-multiple-variables";
 
 import "./variable-chip-list-container.scss";
 
 interface IVariableChipListContainer {
   variables: VariableType[];
 }
 export const VariableChipListContainer = ({ variables }: IVariableChipListContainer) => {
-  const [selectedVariables, setSelectedVariables] = useState<VariableType[]>([]);
+  const { selectedVariables, toggleVariable } = useSelectMultipleVariables();
 
   const onClick = (variable: VariableType) => {
-    let found = false;
-    const newSV = [];
-    selectedVariables.forEach(v => {
-      if (v === variable) {
-        found = true;
-      } else {
-        newSV.push(v);
-      }
-    });
-    if (!found) newSV.push(variable);
-    setSelectedVariables(newSV);
+    toggleVariable(variable);
```

Comment:
Maybe it is better to just drop `onClick` and just pass toggleVariable directly to the component.